Automattic/wp-calypso
# Domains: Improve all domains management performance

Depends on D119262-code.
In an effort to improve the all-domains management page - and make it usable for users with a large number of domains, we're trying some options. 
This PR adds a "light" version of the page - loaded for these users with many domains (we set the threshold to 100). 
It does some optimizations: 
- For the light version:
  - Removes the `Status` column and the ellipsis menu (`Action` column);
  - No site-specific requests are performed (`GET /sites/*`);
  - No dependency on the `sites` data;
  - No dependency on the `purchases` data;
- For all versions:
  - Removes the "Auto-renew" toggle;
  - Doesn't use the `capabilities` state to show the manageable domains - these are now filtered on the backend;
  - Sets the initial sorting to the domain name, ascending;

## Proposed Changes
### Light version
![image](https://github.com/Automattic/wp-calypso/assets/18705930/15c92dd6-78ad-4f83-8c88-35909fd65fae)

### Regular version (unchanged)
![image](https://github.com/Automattic/wp-calypso/assets/18705930/4f7f7308-b65e-4521-bd11-9d7ee94ea438)


## Testing Instructions
- For an account with many domains (> 100), go to `/domains/manage`:
  - Ensure that the page loads - you won't get a timeout anymore
  - Ensure that the page doesn't perform a rerender for each domain - the page doesn't rely on the `sites` state anymore, so no requests to `/sites/*` are performed anymore, thus no rerenders.
  - Ensure that the navigation works and no errors (OOM, timeouts, etc.) happen.
  - Confirm that the notices, "Status" and "Actions" columns are not rendered;
- Ensure that the site domains component is unchanged - and working the same way as it was before;

## Pre-merge Checklist
